eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/ListIterate.java

```diff
@@ -357,6 +358,36 @@ private static boolean nonRandomAccessEquals(List<?> one, List<?> two)
     }
 
     /**
+     * @since 9.1
+     */
+    public static <T, A> MutableList<A> collectWithIndex(
```

Comment:
In a utility like ListIterate is it better to provide only the target collection variant?

Reply:
Good question.  The non-target collection variant is more generally useful when used as utility, which is why I included it.